src/tabs/presets/presets.js

```diff
@@ -465,11 +477,23 @@ presets.getFitPresets = function(searchParams) {
         }
     }
 
-    result.sort((a, b) => (a.priority > b.priority) ? -1 : 1);
+    result.sort((a, b) => this.presetSearchPriorityComparer(a,b));
 
     return result;
 };
 
+presets.presetSearchPriorityComparer = function(presetA, presetB) {
+    if (presetA.lastPickDate && presetB.lastPickDate) {
+        return (presetA.lastPickDate > presetB.lastPickDate) ? -1 : 1;
+    }
+
+    if (presetA.lastPickDate || presetB.lastPickDate) {
+        return (presetA.lastPickDate) ? -1 : 1;
+    }
```

Comment:
Why are we checkin `presetB.lastPickDate` but only using `presetA.lastPickDate` in there?

Reply:
we are in this "IF" in case one of the presets has pickDate and the other does not. But we don't know which does have it and which does not.
This "IF" guarantees that if one of the two presets has lasPickedDate then it will have a priority over the other.